design/one-pager-simple-class-selection.md

```diff
@@ -0,0 +1,411 @@
+
+While not a goal of this proposal, it's worth noting that the label selector
+pattern enables rudimentary scheduling of resource claims to resource classes.
+Class and claim authors _could_ define a labelling schema that allowed claim
+authors to describe the desired properties of their claim using labels. The
+below example illustrates a labelling schema that could be used to describe a
+`RedisCluster` claim that needs a replicated Redis 3.2 instance running on the
+west coast of the USA:
+
+```yaml
+apiVersion: cache.crossplane.io/v1alpha1
+kind: RedisCluster
+metadata:
+  name: my-cache
+spec:
+  classSelector:
+    matchLabels:
+      engineVersion: "3.2"
+      region: us-west
+      replicated: "true"
+```
+
+These resource classes specify that they can satisfy claims for replicated Redis
+3.2 instances on the west coast of the USA via their labels:
+
+```yaml
+apiVersion: cache.gcp.crossplane.io/v1alpha2
+kind: CloudMemorystoreInstanceClass
+metadata:
+  name: gcp-dev
+  labels:
+    engineVersion: "3.2"
+    region: us-west
+    replicated: "true"
+specTemplate:
+  tier: STANDARD_HA
+  region: us-west2
+  memorySizeGb: 1
+  redisVersion: REDIS_3_2
+---
+apiVersion: cache.azure.crossplane.io/v1alpha2
+kind: RedisClass
+metadata:
+  name: azure-dev
+  labels:
+    engineVersion: "3.2"
+    region: us-west
+    replicated: "true"
+specTemplate:
+  resourceGroupName: group-westus-1
+  location: West US
+  sku:
+    name: Basic
+    family: C
+    capacity: 0
+```
+
+Crossplane could (at some future time) solve the specific example above by
+[teaching each resource claim controller] how to match standardised resource
+claim fields to their provider specific resource class fields, but this is true
+only for a very small class of fields that are both highly relevant to claim
+authors and highly likely to translate to every conceivable managed resource
+that could satisfy a claim. Geographic region, node count, and database size are
+examples of such fields.
+
+Label selectors approximate this functionality while it does not yet exist and
+will compliment it when it does. There will likely be characteristics of
+resource classes that claim authors want to match on that cannot be first class
+configurable fields of the resource claim `spec` because they do translate to
+all managed resources.
+
+### Unopinionated Resource Claims
+
+In the absence of spec based scheduling a resource claim that omits the
+`.spec.classSelector` field, as well as the `.spec.classRef` and
+`.spec.resourceRef` fields that allow them to explicitly specify a class or
+existing managed resource respectively will be deemed unopinionated about what
+resource class it needs.
+
+Kubernetes [states that]:
+
+> The semantics of empty or non-specified selectors are dependent on the
+> context, and API types that use selectors should document the validity and
+> meaning of them.
+
+This document proposes that Crossplane fall back to a default resource class - a
```

Comment:
suggest pointing out that class defaults are not cluster-scoped defaults, not per app project namespace as they were with portable classes.

Reply:
Done.